src/main/java/Problem1/InOrderTraverse.java

```diff
@@ -1,12 +1,28 @@
 package Problem1;
 
+
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Stack;
 
 public class InOrderTraverse {
     public static List<Integer> inorderTraversalIterative(TreeNode<Integer> root) {
         // homework
         List<Integer> result = new ArrayList<>();
+        Stack<TreeNode> stack = new Stack();
+
+        TreeNode current = root;
+        while(!stack.isEmpty() || current != null) {
+            if (current != null) {
+                stack.push(current);
+                current = current.left;
+            } else {
+                current = stack.pop();
+                result.add((Integer) current.val);
+                current = current.right;
+            }
+        }
         return result;  // place holder
```

Comment:
you can delete the place holder comment since you are actually using that as code